core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2900,14 +2862,40 @@ private JoinBridgeManager<?> createLookupSourceFactoryManager(
                                 // scale load factor in case partition count (and number of hash build operators)
                                 // is reduced (e.g. by plan rule) with respect to default task concurrency
                                 taskConcurrency / partitionCount));
+
+                context.addDriverFactory(
+                        buildContext.isInputDriver(),
+                        false,
+                        new PhysicalOperation(hashBuilderOperatorFactory, buildSource),
+                        buildContext.getDriverInstanceCount());
+
+                JoinOperatorType joinType = switch (node.getType()) {
```

Comment:
Extract this to separate method, possibly as static `JoinOperatorType` method. If it's the only way we use `innerJoin, probeOuterJoin, ...` then these static constructors don't make much sense

Reply:
Those static constructors are quite handy for testing purposes. I've added another one with join type to handle the generic case